test/ng/compileSpec.js

```diff
@@ -12111,6 +12142,890 @@ describe('$compile', function() {
+            expect(lowercase(element.html())).toEqual('<div>hello</div>');
+          }));
+        });
+      });
+
+    });
+
+    describe('*[style]', function() {
+      it('should NOT set style for untrusted values', inject(function($rootScope, $compile) {
+        element = $compile('<div ng-prop-style="style"></div>')($rootScope);
+        $rootScope.style = 'background: red';
+        expect(function() { $rootScope.$digest(); }).toThrow();
+      }));
+
+      it('should NOT set style for wrongly typed values', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<div ng-prop-style="style"></div>')($rootScope);
+        $rootScope.style = $sce.trustAsHtml('background: red');
+        expect(function() { $rootScope.$digest(); }).toThrow();
+      }));
+
+      it('should set style for trusted values', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<div ng-prop-style="style"></div>')($rootScope);
+        $rootScope.style = $sce.trustAsCss('background: red');
+        $rootScope.$digest();
+        expect(lowercase(element.css('background-color'))).toEqual('red');
+      }));
+    });
+  });
+
+  describe('addPropertySecurityContext', function() {
+    function testProvider(provider) {
+      module(provider);
+      inject(function($compile) { /* done! */});
+    }
+
+    it('should allow adding new properties', function() {
+      testProvider(function($compileProvider) {
+        $compileProvider.addPropertySecurityContext('div', 'title', 'mediaUrl');
+        $compileProvider.addPropertySecurityContext('*', 'my-prop', 'resourceUrl');
+      });
+    });
+
+    it('should allow different sce types of a property on different element types', function() {
+      testProvider(function($compileProvider) {
+        $compileProvider.addPropertySecurityContext('div', 'title', 'mediaUrl');
+        $compileProvider.addPropertySecurityContext('span', 'title', 'css');
+        $compileProvider.addPropertySecurityContext('*', 'title', 'resourceUrl');
+        $compileProvider.addPropertySecurityContext('article', 'title', 'html');
+      });
+    });
+
+    it('should throw \'ctxoverride\' when changing an existing context', function() {
+      testProvider(function($compileProvider) {
+        $compileProvider.addPropertySecurityContext('div', 'title', 'mediaUrl');
+
+        expect(function() {
+          $compileProvider.addPropertySecurityContext('div', 'title', 'resourceUrl');
+        })
+        .toThrowMinErr('$compile', 'ctxoverride', 'Property context \'div.title\' already set to \'mediaUrl\', cannot override to \'resourceUrl\'.');
+      });
+    });
+
+    it('should setting the same property/element to the same value', function() {
+      testProvider(function($compileProvider) {
+        $compileProvider.addPropertySecurityContext('div', 'title', 'mediaUrl');
+        $compileProvider.addPropertySecurityContext('div', 'title', 'mediaUrl');
+      });
+    });
+  });
```

Comment:
We also need tests that verify the effect of `addPropertySecurityContext()` :grin:

Reply:
See fae50665e